Userland/Libraries/LibPDF/Fonts/PS1FontProgram.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022, Julian Offenhäuser <offenhaeuser@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <LibGfx/AffineTransform.h>
+#include <LibGfx/Path.h>
+#include <LibPDF/Error.h>
+
+namespace PDF {
+
+class Reader;
+class Encoding;
+
+class PS1FontProgram : public RefCounted<PS1FontProgram> {
+public:
+    PDFErrorOr<void> parse(ReadonlyBytes const&, size_t cleartext_length, size_t encrypted_length);
+
+    Gfx::Path build_char(u32 code_point, Gfx::FloatPoint const& point, float width);
+
+    RefPtr<Encoding> encoding() const { return m_encoding; }
+
+private:
+    struct Glyph {
+        Gfx::Path path;
+        Gfx::FloatPoint width;
```

Comment:
Does `width` need to be a `Gfx::FloatPoint` just to be able to pass it into `.map()`? To me it would feel more logical if this was a `float` instead.

Reply:
Right, the documentation talks about it being a vector, but we really don't need the y component. Fixed that.